os/include/tinyara/preference.h

```diff
@@ -39,14 +39,6 @@
 #define PREF_PRIVATE_PATH PREF_PATH"/private"
 #define PREF_SHARED_PATH PREF_PATH"/shared"
 
-enum preference_type_e {
-	PREFERENCE_TYPE_INT = 0,
-	PREFERENCE_TYPE_DOUBLE,
-	PREFERENCE_TYPE_BOOL,
-	PREFERENCE_TYPE_STRING,
-	PREFERENCE_TYPE_BINARY,
-};
```

Comment:
Sorry.. I asked you for this change. But now, I think we should keep the same location for every definition. If we keep them at several headers, it could be complex.
(Originally I though we could move entire definitions into the framework. But it's not)

Reply:
We cannot move all definitions to framework, that is true.
Should we close this PR then?

Reply:
How do you think? We have two options.
1. Some definitions are used only in the framework so that they should be defined in the header of the framework.
2. Other almost definitions are in the header of the kernel so that keeping in the kernel header is better to know preference definitions.

Reply:
I think it is OK to have the definition in the module/folder where it is used.
I mean, if we can shift PREFERENCE_TYPEs to framework, but the rest have to stay in kernel, that should also be OK.